docs/RationaleSeq.md

```diff
@@ -106,3 +106,42 @@ information.
 - Initial value: this register is uninitialized. Using an uninitialized value
 results in undefined behavior. We will add an `initialValue` attribute if
 this proves insufficient.
+
+## The Flip Flop Operation
+
+`seq.ff` models a D-Flip Flop operation, with support for various features,
+namely:
+
+- clock edge sensitivity: posedge, negedge or edge-triggered
+- explicit enable signals
+- an optional synchronous & asynchronus reset
+
+This is an abstract general-purpose operation that can be targetted by other
+dialects. It should be further lowered down to either vendor-specific primtives
+or to the SV dialect.
+
+`seq.ff` has the following MLIR format:
+
+```mlir
+%q = seq.ff %input, (posedge|negedge) %clk , %enable, %syncReset, %asyncReset,
+%syncResetValue, %asyncResetValue  : $type(input)
+```
+
+### Rationale
+
```

Comment:
I'm not so concerned about operand format here, but rather how this operand matches up to common flip flop types.  The focus here should be on why this is in the IR and what new modeling or abstraction or transformation opportunities it offers.  Especially what's the use case compared to a seq.reg?

Reply:
My rationale was so that we don't need to do complex analysis to recognize latches vs FF to do latch/FF specific transformations , but I realized it's not that hard to squeeze everything in a single reg op.

I modified this to describe a general-purpose `seq.reg` instaed.